File: momhelp/src/main/java/com/momhelp/service/UserPreferencesService.java
package com.momhelp.service;

import com.momhelp.dto.UserPreferencesRequestDTO;
import com.momhelp.dto.UserPreferencesResponseDTO;

public interface UserPreferencesService {

	UserPreferencesResponseDTO savePreferences(UserPreferencesRequestDTO requestDTO);

	UserPreferencesResponseDTO updatePreferences(Long userId, UserPreferencesRequestDTO requestDTO);

	UserPreferencesResponseDTO getPreferencesByUserId(Long userId);

	void deletePreferences(Long userId);

	UserPreferencesResponseDTO getDefaultPreferences();
}

File: momhelp-frontend/momhelp-frontend/src/services/userPreferencesService.js
import api from './api';

const userPreferencesService = {
  savePreferences: (preferences) => {
    return api.post('/preferences/save', preferences);
  },

  updatePreferences: (userId, preferences) => {
    return api.put(`/preferences/${userId}`, preferences);
  },

  getPreferences: (userId) => {
    return api.get(`/preferences/${userId}`);
  },

  getDefaultPreferences: () => {
    return api.get('/preferences/default');
  },

  deletePreferences: (userId) => {
    return api.delete(`/preferences/${userId}`);
  }
};

export default userPreferencesService;